monai/auto3dseg/seg_summarizer.py

```diff
@@ -0,0 +1,183 @@
+    """
+    SegSummarizer serializes the operations for data analysis in Auto3Dseg pipeline. It loads
+    two types of analyzer functions and execuate differently. The first type of analyzer is
+    CaseAnalyzer which is similar to traditional monai transforms. It can be composed with other
+    transforms to process the data dict which has image/label keys. The second type of analyzer
+    is SummaryAnalyzer which works only on a list of dictionary. Each dictionary is the output
+    of the case analyzers on a single dataset.
+
+    Args:
+        image_key: a string that user specify for the image. The DataAnalyzer will look it up in the
+            datalist to locate the image files of the dataset.
+        label_key: a string that user specify for the label. The DataAnalyzer will look it up in the
+            datalist to locate the label files of the dataset. If label_key is None, the DataAnalyzer
+            will skip looking for labels and all label-related operations.
+        do_ccp: apply the connected component algorithm to process the labels/images
+
+    Examples:
+        .. code-block:: python
+
+            # imports
+
+            summarizer = SegSummarizer("image", "label")
+            transform_list = [
+                LoadImaged(keys=keys),
+                EnsureChannelFirstd(keys=keys),  # this creates label to be (1,H,W,D)
+                ToDeviced(keys=keys, device=device, non_blocking=True),
+                Orientationd(keys=keys, axcodes="RAS"),
+                EnsureTyped(keys=keys, data_type="tensor"),
+                Lambdad(keys="label", func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x),
+                SqueezeDimd(keys=["label"], dim=0),
+                summarizer,
+            ]
+            ...
+            # skip some steps to set up data loader
+            dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=n_workers, collate_fn=no_collation)
+            transform = Compose(transform_list)
+            stats = []
+            for batch_data in dataset:
+                d = transform(batch_data[0])
+                stats.append(d)
+            report = summarizer.summarize(stats)
+    """
+
+    def __init__(self, image_key: str, label_key: str, do_ccp: bool = True) -> None:
+
+        self.image_key = image_key
+        self.label_key = label_key
+
+        self.summary_analyzers: List[OperationType] = []
+        super().__init__()
+
+        self.add_analyzer(FilenameStats(image_key, DataStatsKeys.BY_CASE_IMAGE_PATH), None)
+        self.add_analyzer(FilenameStats(label_key, DataStatsKeys.BY_CASE_LABEL_PATH), None)
+        self.add_analyzer(ImageStats(image_key), ImageStatsSumm())
+
+        if label_key is None:
+            return
+
+        self.add_analyzer(FgImageStats(image_key, label_key), FgImageStatsSumm())
+
+        self.add_analyzer(LabelStats(image_key, label_key, do_ccp=do_ccp), LabelStatsSumm(do_ccp=do_ccp))
```

Comment:
I see you also support other args for these default analyzers, like "average", should we add to the init of this `SegSummarizer`?

Thanks.

Reply:
I added the support for average and do_ccp (connected component analysis) back in commit 754b0b91d8b7fc678c9e92724a1ed0367ebf5dec